packages/components/src/icon/index.js

```diff
@@ -8,24 +8,24 @@ import { cloneElement, createElement, Component, isValidElement } from '@wordpre
  */
 import { Dashicon, SVG } from '../';
 
-function Icon( { icon = null, size, className } ) {
+function Icon( { icon = null, size, className, style } ) {
 	let iconSize;
 
 	if ( 'string' === typeof icon ) {
 		// Dashicons should be 20x20 by default
 		iconSize = size || 20;
-		return <Dashicon icon={ icon } size={ iconSize } className={ className } />;
+		return <Dashicon icon={ icon } size={ iconSize } className={ className } { ...style } />;
 	}
 
 	// Any other icons should be 24x24 by default
 	iconSize = size || 24;
 
 	if ( 'function' === typeof icon ) {
 		if ( icon.prototype instanceof Component ) {
-			return createElement( icon, { className, size: iconSize } );
+			return createElement( icon, { className, style, size: iconSize } );
 		}
 
-		return icon();
+		return icon( { style } );
```

Comment:
In theory we should pass all the props here instead.

Reply:
I thought the same but it didn't have the other props before so didn't want to cause regression

Reply:
Updated with all the props!
I looked over the codebase and could not see a case where this change would break things